### PR TITLE
[18.0][MIG] hr_holidays_public: Migration to 18.0

### PR DESCRIPTION
**Pull request by xaviedoanhduy**
### Change on version 18.0

- `number_of_hours_display` fields has been removed in this `https://github.com/odoo/odoo/pull/156117`

### On the occasion

- I also removed some redundant code in tests (no need to call the denpends function)

### OCA - port

- https://github.com/xaviedoanhduy/hr-holidays/pull/5

### CODE REVIEW

File: hr_holidays_public/__manifest__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 {
     "name": "HR Holidays Public",
-    "version": "17.0.1.0.3",
+    "version": "18.0.1.0.3",
```
Comment (chaule97) [inline hr_holidays_public/__manifest__.py]: `"version": "18.0.1.0.0",`

**Review by chaule97 — COMMENTED** [review]
Update `CREDITS.md`: "The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp."

File: hr_holidays_public/__manifest__.py (hunk 1 above)
Comment (xaviedoanhduy) [reply hr_holidays_public/__manifest__.py]: thank @chaule97, It has been updated

**Review by chaule97 — APPROVED** [review]
Pre-approve: Update CREDITS.md: "The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp."

**Comment by nilshamerlinck** [comment]
where are commits from https://github.com/xaviedoanhduy/hr-holidays/pull/5 ?

**Comment by xaviedoanhduy** [comment]
> where are commits from #5 ?

Hello @nilshamerlinck, I have updated the PR

**Comment by nilshamerlinck** [comment]
did you squash commits already? `https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests`

**Comment by xaviedoanhduy** [comment]
> did you squash commits already? `https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests`

Done. Can you check it agian? 
cc @nilshamerlinck

**Review by thienvh332 — COMMENTED** [review]
Why don't you port this [commit](https://github.com/OCA/hr-holidays/commit/fe15e2c661f1a0f05fdb3fc521deddf2d7b03da3)?

**Comment by xaviedoanhduy** [comment]
> Why don't you port this [commit](https://github.com/OCA/hr-holidays/commit/fe15e2c661f1a0f05fdb3fc521deddf2d7b03da3)?

I only port commits that don't cause conflicts. Otherwise, if conflicts arise, I blacklist them.

**Review by thienvh332 — COMMENTED** [review]
- Why don't you port this [commit](https://github.com/OCA/hr-holidays/commit/13f2713aa558cf54b582492d6c0358aecdec8a84)?

File: hr_holidays_public/models/hr_holidays_public.py [hunk 2]
```diff
@@ -16,7 +15,9 @@ class HrHolidaysPublic(models.Model):
     _order = "year"
 
     display_name = fields.Char("Name", compute="_compute_display_name", store=True)
-    year = fields.Integer("Calendar Year", required=True, default=date.today().year)
+    year = fields.Integer(
+        "Calendar Year", required=True, default=fields.Date.context_today
```
Comment (thienvh332) [inline hr_holidays_public/models/hr_holidays_public.py]: `fields.Date.context_today == default=date.today().year`?

**Comment by nilshamerlinck** [comment]
> I only port commits that don't cause conflicts. Otherwise, if conflicts arise, I blacklist them.

sorry, it's probably because of my unclear guidance: what I meant is that numerous conflicts can be the signal that the missing PR doesn't need to merged because it was already taken into account in another PR, e.g. a migration PR
but if there are some conflicts but the feature is interesting, you can try to adapt the code to keep the feature

File: hr_holidays_public/models/hr_holidays_public.py (hunk 2 above)
Comment (thienvh332) [reply hr_holidays_public/models/hr_holidays_public.py]: `fields.Date.context_today` will return a `datetime.date` type. Not an `integer` type
image

**Comment by xaviedoanhduy** [comment]
hi @thienvh332, missed commits were added manually

File: hr_holidays_summary_email/data/mail_template_data.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                        >${d_to}</td>
+                        </tr>
+                    % endfor
+                    </table>
+                </div>
+            </field>
+            <field name="lang">${object.lang}</field>
+            <field name="auto_delete" eval="True" />
+        </record>
+</odoo>
```
Comment (thienvh332) [inline hr_holidays_summary_email/data/mail_template_data.xml]: Remove it

**Review by thienvh332 — CHANGES_REQUESTED** [review]

File: hr_holidays_summary_email/data/mail_template_data.xml (hunk 3 above)
Comment (xaviedoanhduy) [reply hr_holidays_summary_email/data/mail_template_data.xml]: nice catch, thank you @thienvh332 :)

**Review by thienvh332 — APPROVED** [review]
LGTM

**Pull request closed without merging**